### ISSUE TITLE
Method isThisMyUrl returns false in setup configured with DNS, is there a workaround?

### ISSUE THREAD

**Issue by mgtriffid** [Netflix/eureka#1087]
In our setup Eureka replication is configured following [this guide](https://github.com/Netflix/eureka/wiki/Deploying-Eureka-Servers-in-EC2#configuring-eips-using-dns). DNS records are resolved into AWS ELBs. And looks like this method returns `false` for ELB which actually belongs to stack where this instance is running. Because of that, heartbeats which were sent by clients to particular Eureka instance are replicated to this instance again. And because of that self-preservation can't trigger properly: there are more heartbeats than expected.

Has anybody face this issue? As possible workaround, maybe there is a way to configure instanceInfo.hostName somehow to make it match ELB?

Thanks in advance for advice!

**Comment by pparth** [comment]
Any news here?

**Comment by mgtriffid** [comment]
In our custom build we ended up with following solution.
Each Eureka node generates random UUID, and exposes an endpoint which just returns this UUID.
Custom implementation of `PeerEurekaNodes` overrides method `isThisMyUrl`, and in this method it first calls mentioned endpoint of passed host. And if this call returns the same UUID this node has generated, it means "this is my url", and Eureka remembers that host and always thinks it is it's own host. But I'm not sure how to implement similar functionality in original Eureka, as can't foresee all possible cases. Just sharing an idea here.

**Comment by qiangdavidliu** [comment]
Thanks @mgtriffid for the proposed solution, that would work. In effect what's required is for eureka servers to have an identity that it can match on. When using hostnames directly, this works, but not when servers are fronted by an ELB. The custom changes by @mgtriffid may be something that can be generalised into something that can work with both hostnames and custom UUIDs, and we would welcome such a PR for this project.

**Comment by mgtriffid** [comment]
Submitted a [PR|https://github.com/Netflix/eureka/pull/1101], @qiangdavidliu .

With this PR Eureka still first tries to perform regular check, and only if it fails, we try to fetch peer id.

Also, I've added check only for replications and not for initial resolving peers, because in common scenario (when several instances are starting at the same time) we can't contact peers, as they are not open for traffic yet. Also wanted to perform calls to peers through common mechanism of `EurekaHttpClient`, hence, had to add method to `PeerEurekaNode`, and we don't have instances of `PeerEurekaNode` during initial URLs resolution.

**Comment by holy12345** [comment]
Hi @mgtriffid 
I face the issue which maybe the same to you.
When I start a `eureka server` in local, and start a `eureka client`,  I see the server log has two register log. By the debug the source code , I found `isThisMyUrl` is check not total right

In my mind I think when a `eureka client` start It will register to `eureka server` and `eureka server` will replicate to peer node(In replicate logic, it should skip itself).

But I found it not skip itself : (
